AppInspector.CLI/Program.cs

```diff
@@ -25,103 +26,364 @@ static int Main(string[] args)
+                throw new OpException(MsgHelp.GetString(MsgHelp.ID.PACK_MISSING_OUTPUT_ARG));
+            }
+            else
+            {
+                CommonOutputChecks(options);
+            }
+
+            return RunPackRulesCommand(options);
         }
 
 
-        private static int RunPackRulesCommand(PackRulesCommandOptions opts)
+        private static int VerifyOutputArgsRun(CLIAnalyzeCmdOptions options)
         {
-            Logger logger = Utils.SetupLogging(opts, true);
+            Logger logger = Utils.SetupLogging(options, true);
             WriteOnce.Log = logger;
-            opts.Log = logger;
+            options.Log = logger;
+
+            //analyze with html format limit checks
+            if (options.OutputFileFormat == "html")
+            {
+                if (!string.IsNullOrEmpty(options.OutputFilePath)) //dependent local files won't be there; TODO look into dir copy to target!
+                {
+                    WriteOnce.Info("output file argument ignored for html format");
+                }
+
+                if (options.AllowDupTags) //fix #183; duplicates results for html format is not supported which causes filedialog issues
+
+                {
+                    WriteOnce.Error(MsgHelp.GetString(MsgHelp.ID.ANALYZE_NODUPLICATES_HTML_FORMAT));
+                    throw new OpException(MsgHelp.GetString(MsgHelp.ID.ANALYZE_NODUPLICATES_HTML_FORMAT));
+                }
+
+                if (options.SimpleTagsOnly) //won't work for html that expects full data for UI
+                {
+                    WriteOnce.Error(MsgHelp.GetString(MsgHelp.ID.ANALYZE_SIMPLETAGS_HTML_FORMAT));
+                    throw new Exception(MsgHelp.GetString(MsgHelp.ID.ANALYZE_SIMPLETAGS_HTML_FORMAT));
+                }
+            }
+            else
+            {
+                CommonOutputChecks((CLICommandOptions)options);
+            }
+
+            return RunAnalyzeCommand(options);
+        }
+
+
+        /// <summary>
+        /// Checks that either output filepath is valid or console verbosity is not visible to ensure
+        /// some output can be achieved...other command specific inputs that are relevant to both CLI
+        /// and NuGet callers are checked by the commands themselves
+        /// </summary>
+        /// <param name="options"></param>
+        static void CommonOutputChecks(CLICommandOptions options)
+        {
+            //validate requested format
+            string fileFormatArgs = options.OutputFileFormat;
+            List<string> validFormats = new List<string>();
+            validFormats.Add("json");
+            validFormats.Add("text");
+
+            if (options is CLIAnalyzeCmdOptions)
+            {
+                fileFormatArgs = ((CLIAnalyzeCmdOptions)options).OutputFileFormat;
+            }
+
+            bool isValidFormat = validFormats.Any(v => v.Contains(fileFormatArgs));
```

Comment:
This should probably be .Equals, not .Contains

Reply:
Agreed thanks.  Added a ToLower() as well in case argument was mixed or upper case.

Reply:
Commit added to resolve.